Client/mods/deathmatch/logic/CClientModel.cpp

```diff
@@ -78,30 +78,49 @@ bool CClientModel::Allocate(ushort usParentID)
     return false;
 }
 
+// You can call it only in destructor for DFF.
 bool CClientModel::Deallocate()
 {
     if (!m_bAllocatedByUs)
```

Comment:
Looking over the class...wouldn't it make more sense if `m_bAllocatedByUs = true;` (line 28) was done only in the case where we actually allocated a new model? You won't need that if-block around `pModelInfo->DeallocateModel();`.

Reply:
Some classes CClientPed/CClientVehicle/CClientObject have field `std::shared_ptr<CClientModel> m_clientModel`. I think it is used to make deallocation after unloading a model from stream.